src/legacy/core_plugins/metrics/public/contexts/query_input_bar_context.ts

```diff
@@ -0,0 +1,32 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React from 'react';
+import { Storage } from 'ui/storage';
+import { UiSettingsClientContract } from 'src/core/public';
+
+export interface ICoreSetupContext {
```

Comment:
nit: IMO we don't need an `I` prefix unless it is potentially ambiguous whether the item is an interface/type or a class and we want to differentiate.

Reply:
DOn't we want to be consistent about it?

Reply:
Right now it sounds like platform team has only been using it for situations where it is ambiguous. But also they haven't actually updated everything and may have further recommendations after they go through the process of making these updates.

So for now it probably doesn't matter which approach we choose; we can just wait until a more formal recommendation has been provided and then decide what to do next.